ops/model.py

```diff
@@ -2513,6 +2893,116 @@ def update_relation_data(self, relation_id: int, _entity: 'UnitOrApplication',
                              key: str, value: str):
         self.relation_set(relation_id, key, value, isinstance(_entity, Application))
 
+    def secret_get(self, *,
+                   id: Optional[str] = None,
+                   label: Optional[str] = None,
+                   refresh: bool = False,
+                   peek: bool = False) -> Dict[str, str]:
+        args: List[str] = []
+        if id is not None:
+            args.append(id)
+        if label is not None:
+            args.extend(['--label', label])
+        if refresh:
+            args.append('--refresh')
+        if peek:
+            args.append('--peek')
+        # IMPORTANT: Don't call shared _run_for_secret method here; we want to
+        # be extra sensitive inside secret_get to ensure we never
+        # accidentally log or output secrets, even if _run_for_secret changes.
+        try:
+            result = self._run('secret-get', *args, return_output=True, use_json=True)
+        except ModelError as e:
+            if 'not found' in str(e):
+                raise SecretNotFoundError() from e
+            raise
+        return typing.cast(Dict[str, str], result)
+
+    def _run_for_secret(self, *args: str, return_output: bool = False,
+                        use_json: bool = False) -> Union[str, 'JsonObject', None]:
+        try:
+            return self._run(*args, return_output=return_output, use_json=use_json)
+        except ModelError as e:
+            if 'not found' in str(e):
+                raise SecretNotFoundError() from e
+            raise
+
+    def secret_info_get(self, *,
+                        id: Optional[str] = None,
+                        label: Optional[str] = None) -> SecretInfo:
+        args = []  # type: List[str]
+        if id is not None:
+            args.append(id)
+        if label is not None:
+            args.extend(['--label', label])
+        result = self._run_for_secret('secret-info-get', *args, return_output=True, use_json=True)
+        info_dicts = typing.cast(Dict[str, 'JsonObject'], result)
+        id = list(info_dicts)[0]  # Juju returns dict of {secret_id: {info}}
```

Comment:
`id` is not a great variable name in Python since there is a `id` built-in function

Reply:
Thanks. Yeah, I'm aware, and I generally agree with the rule of thumb that you shouldn't use names of built-in functions for your variable names. However, `id()` is a very rarely-used builtin, and just too good of a name not to use. Here it's just a local variable name, and I wouldn't mind changing that, but particularly when it's a method parameter name (which becomes part of the documented API) like in `Model.get_secret(id=None, label=None)` I think it would be sad to not be able to use `id`. We'd have to use `secret_id`, which would be awkward stuttering: `model.get_secret(secret_id=foo)` instead of the nicer `model.get_secret(id=foo)`.

Python has 70 built-in functions, some of them very rarely used (and they change slowly over time), so I don't think the blanket rule is particularly helpful. For commonly-used built-ins I think the rule of thumb is useful, to avoid confusion, but for rarely-used built-ins I don't think it's a problem.